cache/store.go

```diff
@@ -53,6 +55,8 @@ type Expirable[T any] interface {
 	GetExpiration(object T) (time.Time, error)
 	// HasExpired returns true if the object has expired.
 	HasExpired(object T) (bool, error)
+	// Persist persists the cache to disk.
+	Persist() error
```

Comment:
I'm not sure about this. We want a simple backup/restore mechanism. Users don't have access to the underlying datastore, so they cannot decide on what is being snapshot at a given location.

In the future if we want to enable this, we can add a setter method to change the snapshot location. But I don't see a use case now.

Reply:
That's also one of the reason why `Persist()` was in `Expirable`. You could just decide whether to store or not what was in the cache (based on the ttl and capacity, there still is a possibility to infere what's in there). Having the cache arbitrarily decide to make a snapshot, means we have to prepare for it (writable fs and capacity), and also that leave no choice. In our case we want to snapshot in `IRC` but not in `SC`.

Reply:
That makes it simpler yes, but you want be able to do `p, ok := cache.(Persistable)` to check what type you have in case you receive a cache as a parameter.

> Policies could be persist at close, or persist at every write, or persist at every x interval.
So we would have a single policy to begin. This and the interface are not mutually exclusive.